src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -7,21 +7,25 @@ const SquareMatrix{T} = Union{Diagonal{T}, AbstractTriangular{T}}
 
 function rrule(::typeof(/), A::AbstractMatrix{<:Real}, B::T) where T<:SquareMatrix{<:Real}
     Y = A / B
+    project_A = ProjectTo(A)
+    project_B = ProjectTo(B)
     function slash_pullback(ȳ)
         Ȳ = unthunk(ȳ)
-        ∂A = @thunk Ȳ / B'
-        ∂B = @thunk _unionall_wrapper(T)(-Y' * (Ȳ / B'))
+        ∂A = @thunk project_A(Ȳ / B')
+        ∂B = @thunk project_B(_unionall_wrapper(T)(-Y' * (Ȳ / B')))
```

Comment:
I think the `_unionall_wrapper` might actually be doing the same thing `project` does.
It is worth checking that